src/Cardano/Address/Style/Shelley.hs

```diff
@@ -73,6 +92,35 @@ import qualified Data.ByteArray as BA
 -- - 'HardDerivation': for hierarchical hard derivation of parent to child keys
 -- - 'SoftDerivation': for hierarchical soft derivation of parent to child keys
 --
+-- == Examples
+--
+-- === Generating a root key from 'SomeMnemonic'
+-- > :set -XOverloadedStrings
+-- > :set -XTypeApplications
+-- > :set -XDataKinds
+-- > import Cardano.Mnemonic ( mkSomeMnemonic )
+-- > import Cardano.Address.Derivation ( GenMasterKey(..) )
+-- >
+-- > let (Right mw) = mkSomeMnemonic @'[15] ["network","empty","cause","mean","expire","private","finger","accident","session","problem","absurd","banner","stage","void","what"]
+-- > let sndFactor = mempty -- Or alternatively, a second factor mnemonic transformed to bytes via someMnemonicToBytes
+-- > let rootK = genMasterKeyFromMnemonic mw sndFactor :: Shelley 'RootK XPrv
+--
+-- === Generating an 'Address' from a root key
+--
+-- Let's consider the following 3rd, 4th and 5th derivation paths @0'/0/14@
+--
+--
+-- > import Cardano.Address ( PaymentAddress(..), bech32, mainnetDiscriminant )
+-- > import Cardano.Address.Derivation ( AccountingStyle(..), GenMasterKey(..), toXPub )
+-- >
+-- > let accIx = toEnum 0x80000000
+-- > let acctK = deriveAccountPrivateKey rootK accIx
+-- >
+-- > let addIx = toEnum 0x00000014
+-- > let addrK = deriveAddressPrivateKey acctK UTxOExternal addIx
+-- >
+-- > bech32 $ paymentAddress mainnetDiscriminant (toXPub <$> addrK)
+-- > "addr1vzpfffuj3zkp5g7ct6h4va89caxx9ayq2gvkyfvww48sdncxzgg5v"
```

Comment:
something to investigate  : in cddl we have line like :
```
; 0110: enterprise address: keyhash28 -- 29 bytes -- 47 bech32
```
but I have :
```
"addr1vzpfffuj3zkp5g7ct6h4va89caxx9ayq2gvkyfvww48sdncxzgg5v
```
ie.. `addr` plus `54-characters`  after bech32

Reply:
I wouldn't worry too much about this. In particular, this comment doesn't tell anything about which prefix is used, and the prefix has an influence on the integrated checksum and the encoding itself. What matters is the length of the original data.

Reply:
after recalling bech32 standard it is probably ok, as we have (adding to `addr` + `54-character` rest): 
- addr
- 1 (separator)
- 47-character payload
- 6-character checksum